src/components/shopping-cart/container/index.js

```diff
@@ -1,18 +1,36 @@
+// @flow
+
+import type { DispatchType } from 'constants/redux'
+
 import * as ReactRedux from 'react-redux'
 import * as Reselect from 'reselect'
 
 import selectors from 'redux/selectors'
-import ShoppingCart from '../index'
+import shoppingCartAction from 'redux/actions/shopping-cart'
+
+import * as ShoppingCart from '../index'
+
+type OwnPropsType = {||}
+
+type PropsType = {
+  ...OwnPropsType,
+  ...ShoppingCart.PropsType,
+}
 
 const mapStateToProps = Reselect.createStructuredSelector({
   order: selectors.orders.shoppingCart,
   orderId: selectors.users.shoppingCartOrderId,
   walletAddress: selectors.wallet.address,
 })
 
-const mapDispatchToProps = null
+const mapDispatchToProps = (dispatch: DispatchType) => ({
+  onResetShoppingCart: () => {
+    dispatch(shoppingCartAction.reset())
+  },
+})
 
-export default ReactRedux.connect(
+export default ReactRedux.connect<PropsType, OwnPropsType, _, _, _, _>(
```

Comment:
What is `<PropsType, OwnPropsType, _, _, _, _>` doing?  Does this need a comment for other contributors?

Reply:
no, this is how need to type all Redux `connect` calls